### PR TITLE
Files that enable users to employ the TruFor project in a straightforward manner inside Sherloq

### PR DESCRIPTION
**Pull request by UHstudent**
I've developed some files that should allow users to easily use the TruFor project and use it inside the Sherloq GUI.

User instructions for doing this have been included in the code and will display automatically if someone attempts to use the TruFor tool inside Sherloq, but has not prepped the requirements.

I've made an additional category in the tool list: "AI solutions", because I believe these approaches should have their own category.
I've re-used the icon from the 'various' category as a placeholder.

In the future I'm thinking of adding files so that MM-Fusion can be employed in a similar fashion. I believe this approach is suitable regarding any licenses of the original projects, since we are merely providing a framework which can be used to employ the original work.

Let me know your thoughts and feedback, I can make necessary changes and resubmit a request.

### CODE REVIEW

**Comment by GuidoBartoli** [comment]
Wow, really nice addition, thanks!
Did you check the compatibility of TruFor with TensorFlow 2.16 (the version reported in `requirements.txt`)?

**Comment by UHstudent** [comment]
I found no conflicting issues with TruFor and TensorFlow, but I did forget to check requirements specifically for the TruFor addition. A requirements_ai_solutions.txt file was added, so users can install these dependencies only of they which to extend Sherloq for TruFor. 

I've also made a few finishing touches regarding possible errors and memory issues, which should make the user experience more robust.

If some users could try out these examples and achieve the same output; that would confirm a working solution:

example
image

example1_big
example1_big
example1_big
image
image
image

File: gui/TruFor_main/test_docker/src/analyze_image.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    )
+
+    # Load model
+    if config.TEST.MODEL_FILE:
+        model_state_file = config.TEST.MODEL_FILE
+    else:
+        raise ValueError("Model file is not specified.")
+
+    print(f'=> loading model from {model_state_file}')
+    checkpoint = torch.load(model_state_file, map_location=torch.device(device))
```
[the remark this answers is not in the document]
Comment (UHstudent) [reply gui/TruFor_main/test_docker/src/analyze_image.py:94]: My bad, I had not propperly updated all packages to the correct versions in the requirements file, I've updated this now.
I'm using torch 2.3.0

**Comment by UHstudent** [comment]
@GuidoBartoli yes, that's a good idea. Ideally, some users try out the procedure through my fork and give the go-ahead here. On my end I've succesfully tried it on two machines. Can you maybe test Trufor too?

**Comment by UHstudent** [comment]
Update to note that I've tried the procedure successfully on 3 separate machines without difficulty.

**Pull request merged**